types/stylelint/index.d.ts

```diff
@@ -279,133 +279,340 @@ declare namespace stylelint {
-		readonly 'property-no-unknown': Promise<Rule>;
-		readonly 'property-no-vendor-prefix': Promise<Rule>;
-		readonly 'rule-empty-line-before': Promise<Rule>;
-		readonly 'rule-selector-property-disallowed-list': Promise<Rule>;
-		readonly 'selector-anb-no-unmatchable': Promise<Rule>;
-		readonly 'selector-attribute-name-disallowed-list': Promise<Rule>;
-		readonly 'selector-attribute-operator-allowed-list': Promise<Rule>;
-		readonly 'selector-attribute-operator-disallowed-list': Promise<Rule>;
-		readonly 'selector-attribute-quotes': Promise<Rule>;
-		readonly 'selector-class-pattern': Promise<Rule>;
-		readonly 'selector-combinator-allowed-list': Promise<Rule>;
-		readonly 'selector-combinator-disallowed-list': Promise<Rule>;
-		readonly 'selector-disallowed-list': Promise<Rule>;
-		readonly 'selector-id-pattern': Promise<Rule>;
-		readonly 'selector-max-attribute': Promise<Rule>;
-		readonly 'selector-max-class': Promise<Rule>;
-		readonly 'selector-max-combinators': Promise<Rule>;
-		readonly 'selector-max-compound-selectors': Promise<Rule>;
-		readonly 'selector-max-id': Promise<Rule>;
-		readonly 'selector-max-pseudo-class': Promise<Rule>;
-		readonly 'selector-max-specificity': Promise<Rule>;
-		readonly 'selector-max-type': Promise<Rule>;
-		readonly 'selector-max-universal': Promise<Rule>;
-		readonly 'selector-nested-pattern': Promise<Rule>;
-		readonly 'selector-no-qualifying-type': Promise<Rule>;
-		readonly 'selector-no-vendor-prefix': Promise<Rule>;
-		readonly 'selector-not-notation': Promise<Rule>;
-		readonly 'selector-pseudo-class-allowed-list': Promise<Rule>;
-		readonly 'selector-pseudo-class-disallowed-list': Promise<Rule>;
-		readonly 'selector-pseudo-class-no-unknown': Promise<Rule>;
-		readonly 'selector-pseudo-element-allowed-list': Promise<Rule>;
-		readonly 'selector-pseudo-element-colon-notation': Promise<Rule>;
-		readonly 'selector-pseudo-element-disallowed-list': Promise<Rule>;
-		readonly 'selector-pseudo-element-no-unknown': Promise<Rule>;
-		readonly 'selector-type-case': Promise<Rule>;
-		readonly 'selector-type-no-unknown': Promise<Rule>;
-		readonly 'shorthand-property-no-redundant-values': Promise<Rule>;
-		readonly 'string-no-newline': Promise<Rule>;
-		readonly 'time-min-milliseconds': Promise<Rule>;
-		readonly 'unit-allowed-list': Promise<Rule>;
-		readonly 'unit-disallowed-list': Promise<Rule>;
-		readonly 'unit-no-unknown': Promise<Rule>;
-		readonly 'value-keyword-case': Promise<Rule>;
-		readonly 'value-no-vendor-prefix': Promise<Rule>;
+		readonly [name in keyof CoreRules]: Promise<Rule>;
+	};
+
+	type Primary = string | number | true | RegExp | Array<string | RegExp> | Record<string, any>;
+	type Secondary = Record<string, any>;
+	type CoreRule<P extends Primary, S extends Secondary = any> = Rule<P, S>;
+
+	export type CoreRules = {
+		'alpha-value-notation': CoreRule<
+			'number' | 'percentage',
+			{ exceptProperties: Array<string | RegExp> }
```

Comment:
I thought of it but I thought we would want to find the inconsistencies before doing micro optimisations.
I guess it can be done before just as well.